barretenberg/cpp/src/barretenberg/plonk_honk_shared/relation_checker.hpp

```diff
@@ -24,29 +24,36 @@ template <typename Flavor> class RelationChecker {
 
     /**
      * @brief Check that a single specified relation is satisfied for a set of polynomials
-     *
-     * @tparam Relation a linearly independent Relation to be checked
-     * @param polynomials prover polynomials
-     * @param params a RelationParameters instance
      */
-    template <typename Relation>
-    static void check(const auto& polynomials,
-                      const auto& params,
-                      bool is_linearly_independent,
-                      std::string label = "Relation")
+    template <typename Relation, bool has_linearly_dependent = false>
+    static void check(const auto& polynomials, const auto& params, std::string label = "Relation")
     {
+
         // Define the appropriate accumulator type for the relation and initialize to zero
         typename Relation::SumcheckArrayOfValuesOverSubrelations result;
         for (auto& element : result) {
             element = 0;
         }
 
-        for (size_t i = 0; i < polynomials.w_l.virtual_size(); i++) {
-            if (is_linearly_independent) {
-                // Evaluate each constraint in the relation and check that each is satisfied
-                Relation::accumulate(result, polynomials.get_row(i), params, 1);
-                size_t subrelation_idx = 0;
-                for (auto& element : result) {
+        for (size_t i = 0; i < polynomials.get_polynomial_size(); i++) {
+
+            Relation::accumulate(result, polynomials.get_row(i), params, 1);
+            size_t subrelation_idx = 0;
+
+            // Iterate over all the subrelation results and report if a linearly independent one failed
+            for (auto& element : result) {
+                if constexpr (has_linearly_dependent) {
```

Comment:
Took me a while to see why this logic has to be written this way but I guess its because only relations that have linearly dependent subrelations specify the array SUBRELATION_LINEARLY_INDEPENDENT. This could be simplified quite a bit if all relations provided this. Might be worth doing but doesn't need to be here.